### ISSUE TITLE
KeyError: ('align', 'Anc07')

### ISSUE THREAD

**Issue by lningyawen** [ComparativeGenomicsToolkit/cactus#546]
Hello,

I got a problem, and the error is as follows: 
```
Deprecated toil method.  Please use "toil.lib.conversions.human2bytes()" instead."
Deprecated toil method.  Please use "toil.lib.conversions.human2bytes()" instead."
Deprecated toil method.  Please use "toil.lib.conversions.human2bytes()" instead."
Deprecated toil method.  Please use "toil.lib.conversions.human2bytes()" instead."
Deprecated toil method.  Please use "toil.lib.conversions.human2bytes()" instead."
Deprecated toil method.  Please use "toil.lib.conversions.human2bytes()" instead."
Deprecated toil method.  Please use "toil.lib.conversions.human2bytes()" instead."
Traceback (most recent call last):
  File "/pwd/cactus-bin-v2.0.2/venv/bin/cactus-prepare-toil", line 8, in <module>
    sys.exit(main_toil())
  File "/pwd/cactus-bin-v2.0.2/venv/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 46, in main_toil
    return main(toil_mode=True)
  File "/pwd/cactus-bin-v2.0.2/venv/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 230, in main
    cactusPrepare(options, project)
  File "/pwd/cactus-bin-v2.0.2/venv/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 395, in cactusPrepare
    get_plan(options, project, seqFile, outSeqFile, toil=toil)
  File "/pwd/cactus-bin-v2.0.2/venv/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 529, in get_plan
    fa_promises = [job_idx[("preprocess", dep)].rv() for dep in leaf_deps] + [job_idx[("align", dep)].rv(0) for dep in anc_deps]
  File "/pwd/cactus-bin-v2.0.2/venv/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py", line 529, in <listcomp>
    fa_promises = [job_idx[("preprocess", dep)].rv() for dep in leaf_deps] + [job_idx[("align", dep)].rv(0) for dep in anc_deps]
KeyError: ('align', 'Anc07')
```
I checked the line 529 in `/pwd/cactus-bin-v2.0.2/venv/lib/python3.6/site-packages/cactus/progressive/cactus_prepare.py`. I found it maybe related to the commands I write. However, the command is tested by running Test dataset, the command is as follows:
```
export PATH=/pwd/cactus-bin-v2.0.2/bin:$PATH
export PYTHONPATH=/pwd/cactus-bin-v2.0.2/submodules:$PYTHONPATH
source /pwd/cactus-bin-v2.0.2/venv/bin/activate
export TMPDIR=/workpwd/alignment/tmpdir
export TEMPDIR=/workpwd/alignment/tmpdir

cd /workpwd/alignment/
mkdir workdir outdir tmpdir

export TOIL_TORQUE_ARGS="-q core28"

cactus-prepare-toil  ./jobstore Dragonfly.seqFile \
  --binariesMode local \
  --batchSystem torque \
  --realTimeLogging \
  --outDir /workpwd/alignment/outdir \
  --workDir /workpwd/alignment/workdir \
  --outHal /workpwd/alignment/Dragonfly.hal \
  --disableAutoDeployment \
  --disableCaching \
  --maxNodes 16 \
  --minNodes 1 \
  --preprocessCores 25 \
  --blastCores 25 \
  --alignCores 25 \
  --preprocessMemory 200G \
  --blastMemory 200G \
  --alignMemory 200G \
  --preprocessDisk 2000G \
  --blastDisk 2000G \
  --alignDisk 200G \
  --halAppendDisk 2000G \
  --preprocessPreemptible 50 \
  --blastPreemptible 50 \
  --alignPreemptible 50 \
  --halAppendPreemptible 50 \
  --stats
```
Thank you for any help you can provide.

Yawen

**Comment by glennhickey** [comment]
Hmm, looks like a bug in the scheduler?  Can you share Dragonfly.seqFile so I can try to reproduce?  It doesn't matter if the links to the sequences themselves work or not.

**Comment by lningyawen** [comment]
Hello Glenn

I also realized that my problem this time was in Dragonfly.seqFile for some reason, so I kept trying to modify Dragonfly.seqfile, especially the species tree in Dragonfly.seqfile. When I I tried to run the software with another species tree,  I found that it work. So the structure of the tree affects the running of the software does it?
Here are the two species trees I have used:
`((T_castaneum:0.411163,(D_melanogaster:0.58155,(Z_cesonia:0.146721,T_ni:0.144617):0.356164):0.0767768):0.0387872,(A_mellifera:0.464852,(Z_nevadensis:0.287891,((C_dipterum:0.367016,E_danica:0.273491):0.222599,((tk_seventn:0.122303,(ct_three:0.168034,(((dwq_ten:0.0728138,zq_four:0.0707202):0.0108345,(qk_fiftn:0.152151,wtk_ninetn:0.0606377):0.0141038):0.0655785,(lctk_sixtn:0.0891454,dt_nine:0.0737764):0.0162632):0.0246278):0.0194619):0.0445582,((sck_twenty:0.0824852,zck_fourtn:0.0687664):0.0692472,(bc_six:0.124582,((shanc_seven:0.0889506,ck_eightn:0.0989393):0.0239876,((sc_twelve:0.0844224,bc_eleven:0.108949):0.0111693,((dxc_one:0.0507456,hsc_thirtn:0.0654148):0.0110021,(sc_eight:0.0631128,(xc_two:0.0767545,nsc_five:0.0709204):0.0133184):0.00811877):0.00601582):0.0104055):0.0491556):0.0392768):0.0517244):0.28562):0.059413):0.0754695):0.0387872);` it's ok

`(((((xc_two:0.08697,(sc_eight:0.08622,(nsc_five:0.08574,(hsc_thirtn:0.08575,dxc_one:0.08653):0.00405):0.00228):0.00122):0.00465,(sc_twelve:0.10218,bc_eleven:0.09169):0.00247):0.00432,(shanc_seven:0.09889,ck_eightn:0.10895):0.00470):0.00191,bc_six:0.11137):0.00097,(zck_fourtn:0.10394,sck_twenty:0.10103):0.01276,((((zq_four:0.07709,dwq_ten:0.08002):0.00761,wtk_ninetn:0.07625):0.01227,((tk_seventn:0.09019,ct_three:0.09704):0.00335,(lctk_sixtn:0.09955,dt_nine:0.08446):0.01629):0.00077):0.00224,((qk_sixtn:0.11147,S_curviseta:0.14702):0.00657,(Z_nevadensis:0.14900,((((Z_cesonia:0.11416,C_sasakii:0.12229):0.00087,T_ni:0.12564):0.01252,E_danica:0.10900):0.00209,((T_castaneum:0.14459,(D_melanogaster:0.16790,C_dipterum:0.13310):0.00621):0.00466,A_mellifera:0.09633):0.00337):0.00779):0.00596):0.00707):0.00965);` Using this tree to run cactus will report an error.

So how does the species tree affect the running of the cactus?

Thank you so much for your help!!!
Best, Yawen
